Nos/Service/SocialGraph.swift

```diff
@@ -0,0 +1,178 @@
+//
+//  SocialGraph.swift
+//  Nos
+//
+//  Created by Matthew Lorentz on 4/18/23.
+//
+
+import Foundation
+import CoreData
+import Logger
+
+/// A representation of the people a given user follows and the people they follow designed to cache this data in 
+/// memory and make it fast to access. This class watches the database for changes to the social graph and updates 
+/// itself accordingly.
+@MainActor class SocialGraph: NSObject, ObservableObject, NSFetchedResultsControllerDelegate {
```

Comment:
Good idea! This is something we should have made in Planetary. It has to be useful.